packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -2680,6 +2644,1144 @@ test.serial('writeGranuleFromApi() removes preexisting granule file from postgre
+    knex,
+    granuleModel,
+  });
+
+  t.true(await granuleModel.exists({ granuleId: completeGranule.granuleId }));
+  t.true(
+    await granulePgModel.exists(knex, {
+      granule_id: completeGranule.granuleId,
+      collection_cumulus_id: collectionCumulusId,
+    })
+  );
+  t.true(await esGranulesClient.exists(completeGranule.granuleId));
+
+  const initialDynamoGranuleRecord = await granuleModel.get({ granuleId });
+
+  // Create new files for granule that will not update. These records will exist in database
+  // during subsequent granule write from message
+  const failUpdateFiles = [
+    fakeFileFactory({ size: 5 }),
+    fakeFileFactory({ size: 10 }),
+    fakeFileFactory({ size: 15 }),
+    fakeFileFactory({ size: 20 }),
+  ];
+
+  const failUpdateGranule = {
+    ...completeGranule,
+    files: failUpdateFiles,
+    status: 'running',
+    cmrLink: 'updatedGranuled.com',
+  };
+  cumulusMessage.meta.status = 'running';
+  cumulusMessage.payload.granules[0] = failUpdateGranule;
+
+  await writeGranulesFromMessage({
+    cumulusMessage,
+    executionCumulusId,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  const dynamoRecord = await granuleModel.get({ granuleId });
+  const postgresRecord = await granulePgModel.get(knex, {
+    granule_id: granuleId,
+    collection_cumulus_id: collectionCumulusId,
+  });
+  const apiFormattedPostgresGranule
+    = await translatePostgresGranuleToApiGranule({
+      granulePgRecord: postgresRecord,
+      knexOrTransaction: knex,
+    });
+  const esRecord = await esGranulesClient.get(granuleId);
+
+  const expectedGranule = {
+    ...initialDynamoGranuleRecord,
+    // should already be part of initialDynamoGranuleRecord
+    // but setting more explicitly for test transparency
+    // these values and files were not changed so have values from the first write
+    files: files,
+    status: 'completed',
+    cmrLink: 'example.com',
+    productVolume: '15',
+    createdAt: initialDynamoGranuleRecord.createdAt,
+    timestamp: initialDynamoGranuleRecord.timestamp,
+    updatedAt: initialDynamoGranuleRecord.updatedAt,
+  };
+
+  // Files array order is not promised to match between datastores
+  [esRecord, dynamoRecord, expectedGranule, apiFormattedPostgresGranule].forEach((record) => {
+    record.files.sort((f1, f2) => sortFilesByBuckets(f1, f2));
+  });
+
+  // Translated postgres granule matches expected updatedGranule
+  // minus model defaults
+  t.deepEqual(
+    apiFormattedPostgresGranule,
+    removeNilProperties(expectedGranule)
+  );
+  t.deepEqual(dynamoRecord, removeNilProperties(expectedGranule));
+  t.deepEqual(omit(esRecord, ['_id']), removeNilProperties(expectedGranule));
+});
+
+test.serial('writeGranulesFromMessage() on update changing granule status to "queued", with different files and the same execution, does not update the granule values or files in Postgres/DynamoDB/ES, so the pre-existing values and files will persist', async (t) => {
```

Comment:
Same question as above. How do we get into this situation?

Reply:
I think also to @npauzenga 's comment, with our use of queue granules that shouldn't be happening, but it is *possible* for this to occur in this method, and therefore a corner we should test in case of upstream changes.